server/src/main/java/org/opensearch/index/remote/RemoteStoreEnums.java

```diff
@@ -103,9 +103,24 @@ boolean requiresHashAlgorithm() {
         HASHED_PREFIX(1) {
             @Override
             public BlobPath generatePath(PathInput pathInput, PathHashAlgorithm hashAlgorithm) {
-                // TODO - We need to implement this, keeping the same path as Fixed for sake of multiple tests that can fail otherwise.
-                // throw new UnsupportedOperationException("Not implemented"); --> Not using this for unblocking couple of tests.
+                assert Objects.nonNull(hashAlgorithm) : "hashAlgorithm is expected to be non-null";
+                return addPrefix(pathInput.basePath(), hashAlgorithm.hash(pathInput)).add(pathInput.indexUUID())
+                    .add(pathInput.shardId())
+                    .add(pathInput.dataCategory().getName())
+                    .add(pathInput.dataType().getName());
+            }
+
+            @Override
+            boolean requiresHashAlgorithm() {
+                return true;
+            }
+        },
+        HASHED_INFIX(2) {
```

Comment:
What is the use of `code` here?

Reply:
We are using the code equivalent during serDe. This is also what is being persisted in the remote shallow snapshot metadata file.

Reply:
My concern is mostly with readability of the enum value while debugging something. If the value is not something a developer/user can see, I am fine with the numbered code.